### PR TITLE
Improve player controller

### PR DESCRIPTION
**Pull request by Yonimevicio**

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-jzb3kv Improve player controller script](https://app.clickup.com/t/jzb3kv)

**Review by isaacfdez — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

File: Penteract/Assets/Scripts/PlayerController.cpp [hunk 1]
```diff
@@ -245,17 +190,80 @@ void PlayerController::SwitchCharacter() {
...
+		shootCooldownRemaing = shootCooldown;
+		shooting = true;
+		if (fang->IsActive()) {
+			fangCompParticle->Play();
+		}
+		else {
+			onimaruCompParticle->Play();
+		}
 
+		float3 start = transform->GetPosition();
```
Comment (Jordi-Gil) [inline Penteract/Assets/Scripts/PlayerController.cpp:221]: Maybe here you want the global position?

File: Penteract/Assets/Scripts/PlayerController.cpp [hunk 2]
```diff
@@ -341,49 +382,49 @@ void PlayerController::PlayAnimation(MovementDirection md, bool isFang) {
...
+		hudControllerScript->UpdateCooldowns(0.0f, 0.0f, 0.0f, realDashCooldown, 0.0f, 0.0f, realSwitchCooldown);
 
+		if (lifePointsFang <= 0 || lifePointsOni <= 0) {
+			SceneManager::ChangeScene("Assets/Scenes/LoseScene.scene");
+		}
 	}
+}
 
+void PlayerController::UpdateCameraPosition() {
+	cameraTransform->SetPosition(float3(transform->GetGlobalPosition().x + cameraOffsetX,
```
Comment (Jordi-Gil) [inline Penteract/Assets/Scripts/PlayerController.cpp:414]: Maybe here you want to set the Global Position

File: Penteract/Assets/Scripts/PlayerController.cpp (hunk 1 above)
Comment (Yonimevicio) [reply Penteract/Assets/Scripts/PlayerController.cpp:221]: yeah it is correct, i will fix it

**Review by isaacfdez — APPROVED** [review]

**Review by Jordi-Gil — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Pull request merged**